### DISCUSSION TITLE
Send command to the Webapi device

### DISCUSSION THREAD

**Discussion by alkinoy** [frangoteam/FUXA#1297] [Q&A]
Hello here!

Thanks for this awesome application! Probably my question already answered, but unfortunately I didn't find it :(

I have my own server which was added to FUXA project as Webapi device. And I successfully see my data on the dashboard.
Now I want to add some action which can be pushed to my API server as a command from dashboard. For example I want to add possibility to press button on the dashboard and got API call to my server.

How can I do it?

Currently the only way I see - add script which will perform POST request to the my API server. And link this script to the control event. 

But not sure it is possible or is correct way. Can you advise any other solution?

Thanks, Oleksiy.

**Comment by unocelli** [comment]
Hi, there is the option where you can see the API server as device #650

**Reply by alkinoy** [reply]
@unocelli Umberto, thanks for reply.

It could be a way. In my case it is a little bit hard to do. I've installed docker based version of the FUXA, so I have  not direct access to the envParams.js... Also as I see it will switch whole app into dev mode.

Is it possible to send request with script? I've tried with fetch() but seems it stuck during request preparing
![image](https://github.com/user-attachments/assets/fbca9560-cb2a-4c45-88db-a3f02ec06cab)

Same case with XMLHttpRequest
![image](https://github.com/user-attachments/assets/03bcb998-91e7-42fa-b030-38b01e0d45e2)


UPD: the reason is `>fetch is not defined`
![image](https://github.com/user-attachments/assets/ee5f4eb6-2e6c-4169-a8fc-163f686fdd0b)
same for the example with XMLHttpRequest
Regards, Oleksiy.

**Reply by alkinoy** [reply]
ooookay, looks like it works

```
try {
  const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
  fetch("http://127.0.0.1/api/tags", {
    method: "POST",
    headers: {'Content-Type': 'application/json'}
  }).then(res => {
    console.log("Request complete! response:", res);
  });
} catch (err) {
  console.log(err.message);
}
```
